recipes/funchook/all/conanfile.py

```diff
@@ -0,0 +1,114 @@
+
+    def package_id(self):
+        self.info.shared_library_package_id()
+
+    def requirements(self):
+        if self.options.disassembler == "capstone":
+            self.requires("capstone/4.0.2")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def validate(self):
+        if self.options.disassembler == "zydis":
+            raise ConanInvalidConfiguration("disassembler 'zydis' currently not supported by this Conan recipe")
+        if 'arm' in str(self.settings.arch):
+            if self.settings.os == "Macos":
+                raise ConanInvalidConfiguration("Funchook does not support ARM on MacOS")
+            if self.options.disassembler != "capstone":
+                raise ConanInvalidConfiguration("disassembler must be 'capstone' for arm arch")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["FUNCHOOK_BUILD_TESTS"] = False
+        self._cmake.definitions["FUNCHOOK_BUILD_SHARED"] = self.options.shared
+        self._cmake.definitions["FUNCHOOK_BUILD_STATIC"] = not self.options.shared
+        if self.options.disassembler == "diStorm3":
+            self._cmake.definitions["FUNCHOOK_DISASM"] = "distorm"
+        else:
+            self._cmake.definitions["FUNCHOOK_DISASM"] = self.options.disassembler
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+        self.copy("*.h", dst="include", src=(self._source_subfolder + "/include"))
+        self.copy("*.lib", dst="lib", keep_path=False)
+        self.copy("*.dll", dst="bin", keep_path=False)
+        if self.options.shared:
+            self.copy("*.so*", dst="lib", keep_path=False)
+            self.copy("*.dylib*", dst="lib", keep_path=False)
+        else:
+            self.copy("*.a", dst="lib", keep_path=False)
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
+
+        # in case of static linking, funchook needs to be at the start (before other .a files)
+        funchook_libs = [i for i in self.cpp_info.libs if i.startswith('funchook')]
+        self.cpp_info.libs = [i for i in self.cpp_info.libs if i not in funchook_libs]
+        self.cpp_info.libs = funchook_libs + self.cpp_info.libs
```

Comment:
```suggestion
        self.cpp_info.libs = ["funkhook"]
```

Prefer explicit names. collect_libs will sort by alphabetic order, it's not safe.

Reply:
Yeah, so I had shied away from explicit names because diStorm3.[a/dll/so/lib] was sometimes present and other times not. 

But!  As discussed above this should be its own recipe anyway.  So I think that solves my problem!  Will change this as part of splitting the recipes...

Reply:
Thank you!